### ISSUE TITLE
Data lost when using FastField + "Enter-to-submit"

### ISSUE THREAD

**Issue by leesiongchan** [jaredpalmer/formik#577]
## Bug, Feature, or Question?
Bug

## Current Behavior
Data lost when submit using `Enter` key.

## Desired Behavior
Should save data before submitting.

## Additional Information

---

- Formik Version: 1.0.0-alpha.4
- React Version: 16.3
- TypeScript Version:
- CodeSandbox Link: https://codesandbox.io/s/lxyxol8j99
- OS: 
- Node Version:
- Package Manager and Version: yarn

**Comment by jaredpalmer** [comment]
Hrmmmm. I wonderrr if we need to add an enter key press listener to auto blur the field. Right now fast field only sends up its local state to the parent if onBlur is called. Anyone else have a suggested solution?

**Comment by jaredpalmer** [comment]
Okay yes I confirmed the described behavior just now on my iPhone. Thank you for the sandbox. Very helpful.

**Comment by lookfirst** [comment]
possibly related: https://github.com/jaredpalmer/formik/issues/597

**Comment by jaredpalmer** [comment]
I think i have the solution: https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/activeElement

It would be dom only, but we could get the active element value when the enter key is pressed.

**Comment by ernestopye** [comment]
Somewhat unrelated - thoughts on not only updating form state on field blur, but perhaps have it debounced or throttled from FastField? FastField only updating as needed is awesome, but sometimes the UX isn't great when you have to click away from the field to know certain options as available (due to them requiring a re-render to update).

For example save/reset buttons with `disabled={!dirty}` would be disabled until you lost focus after changing at least one field, at which point "dirty" is set and the buttons re-render. 

I know it doesn't fully solve this specific issue. I ran into this exact same problem with some components I wrote for work. It made the issue less noticeable however, as you'd have to press enter pretty quickly after typing something for it to break. 😄

Let me know your thoughts. I'd like to help if possible.

**Comment by ericbiewener** [comment]
Currently trying a solution that uses a data attribute to communicate to `onSubmit` whether or not to modify the values:

```js
const { activeElement} = document;
if (activeElement instanceof HTMLInputElement && activeElement.dataset.fastField) {
  values = {
    ...values,
    [activeElement.name]: activeElement.value,
  }
}
```

**Comment by gustavopch** [comment]
Any news on this?

**Comment by jaredpalmer** [comment]
This is fixed in 1.2.
